junit/src/main/java/cucumber/runtime/junit/IdProvider.java

```diff
@@ -0,0 +1,56 @@
+package cucumber.runtime.junit;
+
+import java.io.Serializable;
+
+/**
+ * JUnit Descriptions require an unique id. However this id should be identical between test
+ * re-runs. This allows descriptions to be used to determine which failed tests should be rerun.
+ *
+ * IdProvider generates predictable unique ids. The feature name is used as a base to disambiguate
+ * between different features.
+ */
+class IdProvider {
+
+    private long id = 0;
+    private final String base;
+
+    IdProvider(String base) {
+        this.base = base;
+    }
+
+    Serializable next() {
+        return new Id(base, id++);
```

Comment:
Using a sequence counter as a base for the id-object makes them unpredictable. As I understand it using
```
cucumber/examples/java/calculator/basic_arithmetic.feature:7:12
```
and
```
cucumber/examples/java/calculator/basic_arithmetic.feature:12
```
would result in different ids for the test case from the scenario on line 12 (feature file from the [java-calculator](https://github.com/cucumber/cucumber-jvm/tree/master/examples/java-calculator) example in the repo).

The id should instead be base on the uri(/path) of the feature file and the line number of the feature/scenario/etc.